src/mapping/impl/BisectionRBFTuner.hpp

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+#include <mapping/MathHelper.hpp>
+#include <mapping/config/MappingConfigurationTypes.hpp>
+#include <mesh/Mesh.hpp>
+#include "mapping/impl/RBFParameterTuner.hpp"
+
+namespace precice::mapping {
+
+template <typename Solver>
+class BisectionRBFTuner : public RBFParameterTuner<Solver> {
```

Comment:
I look at this code now 3 times and think that I get a vague grasp on what you are trying to achieve here. I have a feeling that you attempted CRTP here.

Assuming that there used to be another tuner, this still isn't CRTP. It's a refactoring of common classes into a base. This is fine, but the base shouldn't be templatized here.
In the current state, the RBFParameterTuner can just be merged into this class.

The templatization of this class is also not necessary. Some places require an RBF or a Solver. If you want to ensure consistent RBF usage, then you can use the RBF as a template parameter, then templatize the functions that need a solver type and ensure that it uses the same RBF.

In general the BisectionRBFTuner and RBFParameterTuner need simplification.